host-controller/src/main/java/org/jboss/as/host/controller/mgmt/HostControllerRegistrationHandler.java

```diff
@@ -332,9 +354,15 @@ public void execute(final OperationContext context, final ModelNode operation) t
             }
             // Remotely resolve the subsystem versions and create the transformation
             registrationContext.processSubsystems(transformers, extensions);
-            // Now run the read-domain model operation
-            final ReadMasterDomainModelHandler handler = new ReadMasterDomainModelHandler(hostInfo, transformers, domainController.getExtensionRegistry());
-            context.addStep(READ_DOMAIN_MODEL.getOperation(), handler, OperationContext.Stage.MODEL);
+
+            final Transformers.ResourceIgnoredTransformationRegistry ignoredTransformationRegistry;
+            final Resource resource = context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS);
+            final ReadMasterDomainModelUtil.RequiredConfigurationHolder rc = hostInfo.populateRequiredConfigurationHolder(resource, domainController.getExtensionRegistry());
+            ignoredTransformationRegistry = ReadMasterDomainModelUtil.createHostIgnoredRegistry(hostInfo, rc);
+
+            final ReadMasterDomainModelUtil readUtil = ReadMasterDomainModelUtil.readMasterDomainResourcesForInitialConnect(transformers,
+                    transformationInputs, ignoredTransformationRegistry, transformationInputs.getRootResource());
+            context.getResult().set(readUtil.getDescribedResources());
```

Comment:
Yeah, can definitely do that, those methods were public, so I wasn't so sure how easily changed they were.